plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevControllerExtensions.kt

```diff
@@ -135,7 +148,57 @@ suspend fun FeatureDevController.onCodeGeneration(
 
         messenger.sendSystemPrompt(tabId = tabId, followUp = getFollowUpOptions(session.sessionState.phase, InsertAction.ALL))
 
+        session.sendMetricDataTelemetry(
```

Comment:
Instead of setting success here, should we set it at the very end of the method, right before the method closes? That would allow us to validate that the remaining actions to modify the UI complete successfully.